src/Categories/Bicategory/Construction/Bimodules/Tensorproduct/Associator.agda

```diff
@@ -0,0 +1,1119 @@
+
+      F₃ ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ actionˡ₃ ◁ (F₂ ∘₁ F₁)
+        ∘ᵥ associator.to
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ sym-assoc₂ ⟩
+
+      (F₃ ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ actionˡ₃ ◁ (F₂ ∘₁ F₁))
+        ∘ᵥ associator.to
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ ◁-▷-exchg ⟩∘⟨refl ⟩
+
+      (actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ (F₃ ∘₁ T₃) ▷ Coequalizer.arr F₂⊗F₁)
+        ∘ᵥ associator.to
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ assoc₂ ⟩
+
+      actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ (F₃ ∘₁ T₃) ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ associator.to
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ refl⟩∘⟨ sym-assoc₂ ⟩
+
+      actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ ((F₃ ∘₁ T₃) ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ associator.to)
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ refl⟩∘⟨ ⟺ α⇐-▷-∘₁ ⟩∘⟨refl ⟩
+
+      actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ (associator.to
+        ∘ᵥ F₃ ▷ T₃ ▷ Coequalizer.arr F₂⊗F₁)
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ refl⟩∘⟨ assoc₂ ⟩
+
+      actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ associator.to
+        ∘ᵥ F₃ ▷ T₃ ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ≈⟨ sym-assoc₂ ⟩
+
+      (actionˡ₃ ◁ Coequalizer.obj F₂⊗F₁
+        ∘ᵥ associator.to)
+        ∘ᵥ F₃ ▷ T₃ ▷ Coequalizer.arr F₂⊗F₁
+        ∘ᵥ F₃ ▷ associator.from
+        ∘ᵥ associator.from                     ∎
+
+      where
+        open hom.HomReasoning
+  -- end abstract --
+  
+  coeqcoeqᶠⁱ : Coequalizer f⇒i₁ f⇒i₂
+  coeqcoeqᶠⁱ = [F₃⊗F₂]⊗F₁
+  
+  coeqcoeqᵍʰ : Coequalizer g⇒h₁ g⇒h₂
+  coeqcoeqᵍʰ = F₃⊗[F₂⊗F₁]
+
+  
+
+  Associator⊗Iso : Bimodule.F ((B₃ ⊗₀ B₂) ⊗₀ B₁) ≅ Bimodule.F (B₃ ⊗₀ B₂ ⊗₀ B₁)
+  Associator⊗Iso = CoeqsAreIsomorphic
+                    coeqᶠ coeqᵍ coeqʰ coeqⁱ
+                    f⇒i₁ f⇒i₂ g⇒h₁ g⇒h₂
+                    sq₁ᶠⁱ sq₂ᶠⁱ sq₁ᵍʰ sq₂ᵍʰ
+                    coeqcoeqᵍʰ coeqcoeqᶠⁱ
+                    
+  α⇒⊗ : (Bimodule.F ((B₃ ⊗₀ B₂) ⊗₀ B₁)) ⇒₂ (Bimodule.F (B₃ ⊗₀ B₂ ⊗₀ B₁))
+  α⇒⊗ = _≅_.from Associator⊗Iso
+
+  hexagon : Coequalizer.arr F₃⊗[F₂⊗F₁] ∘ᵥ F₃ ▷ Coequalizer.arr F₂⊗F₁ ∘ᵥ associator.from
+             ≈ α⇒⊗ ∘ᵥ Coequalizer.arr [F₃⊗F₂]⊗F₁ ∘ᵥ Coequalizer.arr F₃⊗F₂ ◁ F₁
+  hexagon = IsoFitsInPentagon
+                coeqᶠ coeqᵍ coeqʰ coeqⁱ
+                f⇒i₁ f⇒i₂ g⇒h₁ g⇒h₂
+                sq₁ᶠⁱ sq₂ᶠⁱ sq₁ᵍʰ sq₂ᵍʰ
+                coeqcoeqᵍʰ coeqcoeqᶠⁱ
+    where
+      open CoequalizerOfCoequalizer using (IsoFitsInPentagon)
+
+open 2-cell using (α⇒⊗; hexagon) public
+
+module Linear-Left where
+  open Bimodule B₁ using () renaming (actionˡ to actionˡ₁)
+
+  abstract
+    linearˡ∘arr∘arr : ((Bimodule.actionˡ (B₃ ⊗₀ B₂ ⊗₀ B₁)
```

Comment:
Is the 'shape' of this lemma forced? It's always odd when the first N steps (and last K steps) are all assoc-shuffling.